.github/workflows/pathogen-repo-ci.yaml

```
@@ -44,10 +44,47 @@ on:
         default: ""
         required: false
 
+permissions:
+  contents: read
+  packages: read
+
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
+      # Log in, if possible, to docker.io (Docker Hub), since authenticated
+      # requests get higher rate limits (e.g. for image pulls).  Our org-level
+      # secret DOCKER_TOKEN_PUBLIC_READ_ONLY is available to all our public
+      # repos on GitHub but only available here to this reusable workflow when
+      # called with "secrets: inherit".  On Docker Hub, the token is granted
+      # "public read-only" access.
+      - id: DOCKER_TOKEN_PUBLIC_READ_ONLY
+        name: Check if DOCKER_TOKEN_PUBLIC_READ_ONLY secret is available
+        env:
+          DOCKER_TOKEN_PUBLIC_READ_ONLY: ${{ secrets.DOCKER_TOKEN_PUBLIC_READ_ONLY }}
+        run: |
+          tee -a "$GITHUB_OUTPUT" <<<available=${DOCKER_TOKEN_PUBLIC_READ_ONLY:+yes}
+
+      - if: steps.DOCKER_TOKEN_PUBLIC_READ_ONLY.outputs.available == 'yes'
+        name: Log in to docker.io
+        uses: docker/login-action@v2
+        with:
+          registry: docker.io
+          username: nextstrainbot
+          password: ${{ secrets.DOCKER_TOKEN_PUBLIC_READ_ONLY }}
+        continue-on-error: true
```

Is this an alternative?

```suggestion
      - env:
          DOCKER_TOKEN_PUBLIC_READ_ONLY: ${{ secrets.DOCKER_TOKEN_PUBLIC_READ_ONLY }}
        if: env.DOCKER_TOKEN_PUBLIC_READ_ONLY != ''
        name: Log in to docker.io
        uses: docker/login-action@v2
        with:
          registry: docker.io
          username: nextstrainbot
          password: ${{ secrets.DOCKER_TOKEN_PUBLIC_READ_ONLY }}
        continue-on-error: true
```

---

Hmm! Maybe? The `env` context is available in `jobs.<job-id>.steps.if`, but it's not clear to me if the step env is applied by that point or if it only includes the job and workflow env. We could test it!

---

The docs don't seem to mention it, but the [suggestion](https://docs.github.com/en/actions/security-guides/encrypted-secrets#using-encrypted-secrets-in-a-workflow) is to use job env not step env for `step.if`:

> Instead, consider setting secrets as job-level environment variables, then referencing the environment variables to conditionally run steps in the job.

---

Heh, I have a test but can't test it until GitHub Actions stops having its outage. :upside_down_face:

---

https://github.com/nextstrain/.github/pull/41